### ISSUE TITLE
Smarter dataset splitting

### ISSUE THREAD

**Issue by ivanzvonkov** [nasaharvest/crop-mask#21]
# Issue
Currently, the train, validation, and test set split is done [randomly](https://github.com/nasaharvest/crop-mask/blob/b485b56c9d9140f4ac7be906047f4eaf2b6600e8/src/ETL/engineer.py#L233). Highly correlated data points between the training and validation/test set can lead to an inflated validation/test set score. 

# Example
Consider 80 data points from Busia, Kenya, and 20 data points from Nairobi, Kenya. 
A random dataset split will most likely yield a training, validation/test set where each set contains data points from both regions. Example split: training set (64 Busia, 16 Nairobi), validation/test set (16 Busia, 4 Nairobi). 

Since the model was trained on many points from Busia it will most likely make good predictions for that region and therefore result in a high validation/test set score. 

However, the purpose of the validation/test score is to give an estimate of how well the trained model will perform on out-of-sample data. The above split does not fulfill that purpose because most of the validation/test set data is highly correlated to the training set data. 

Splitting based on the region: training set (80 Busia), validation/test set (20 Nairobi) may better fulfill that purpose because the validation/test set score will better represent how well the model performs on out of sample data.

# To do

- [ ] Review existing work on dataset splits with satellite data (if there is any)
- [ ] Brainstorm dataset splitting strategies
     - [ ] Latitude and longitude (data points that are close together go in the same set)
     - [ ] Dataset source (data points from the same source dataset go in the same set)
     - [ ] Temporal dimension (data points from the same year/season go in the same set)
- [ ] Implement dataset splitting strategies
- [ ] Obtain labeled dataset for a specific country and a separate test set that was used to evaluate the model
- [ ] Train several new models using different dataset splitting strategies 
- [ ] Compare validation and test set scores on the labeled dataset to the score on the separate test set
